### PR TITLE
Add ULM build integration

### LINKED ISSUE #14: Add ULM build integration

### PR DESCRIPTION
**Pull request by sskeirik**
Several changes:

1. there is a new K WASM main file that imports the ULM hooks 
2. the makefile now contains targets to build ULM-integrated WASM in a separate build directory
3. the git ignore file was updated to exclude the build directory

closes: #14

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -29,6 +29,116 @@ clean: pykwasm
...
+
+ULM_HOOKS_CLONE_DIR=$(ULM_DEP_DIR)/ulm
+ULM_HOOKS_DIR=$(ULM_HOOKS_CLONE_DIR)/kllvm
+ULM_HOOKS_SRC=ulm_kllvm.cpp ulm_hooks.cpp ulm_kllvm_c.cpp
+ULM_HOOKS_LIB=libulmkllvm.so
+ULM_HOOKS_TARGET=$(ULM_LIB_DIR)/$(ULM_HOOKS_LIB)
+
+### ULM Plugin
+
+$(ULM_PLUGIN_DIR)/.git:
```
Comment (virgil-serbanuta) [inline Makefile]: I'm not sure if it's worth renaming, but I usually think of this plugin as "the crypto plugin", not "the ulm plugin".

File: Makefile [hunk 2]
```diff
@@ -29,6 +29,116 @@ clean: pykwasm
...
+
+.PHONY: ulm-plugin-build
+ulm-plugin-build: $(ULM_PLUGIN_TARGET)
+
+### ULM Hooks
+
+$(ULM_HOOKS_CLONE_DIR)/.git:
+	@mkdir -p $(ULM_DEP_DIR)
+	cd $(ULM_DEP_DIR); \
+	  git clone https://github.com/pi-squared-inc/ulm
```
Comment (virgil-serbanuta) [inline Makefile]: Do you need the entire repository, or would `--depth=1` be better?

File: Makefile [hunk 3]
```diff
@@ -29,6 +29,116 @@ clean: pykwasm
...
+
+$(ULM_HOOKS_TARGET): | $(ULM_HOOKS_CLONE_DIR)/.git
+	@mkdir -p $(ULM_LIB_DIR)
+	cd $(ULM_HOOKS_DIR); \
```
Comment (virgil-serbanuta) [inline Makefile:86]: Is this the same as `make -C $(ULM_HOOKS_DIR) $(ULM_HOOKS_LIB_NAME)`?

Comment (sskeirik) [reply Makefile:86]: So, I found that the upstream ULM hardcoded clang++ 16, which I didn't have on my machine, so I replaced it with a generic C++ compiler. Thanks for pointing this out --- this comment did apply to a different make invocation.

File: Makefile (hunk 2 above)
Comment (sskeirik) [reply Makefile]: I updated this as you suggested.

File: Makefile (hunk 1 above)
Comment (sskeirik) [reply Makefile]: I updated these makefile target variable names to use the name "krypto."

**Review by virgil-serbanuta — APPROVED** [review]

**Pull request merged**